language/move-stdlib/src/natives/string.rs

```diff
@@ -190,6 +190,45 @@ pub fn make_native_index_of(gas_params: IndexOfGasParameters) -> NativeFunction
     )
 }
 
+/***************************************************************************************************
+ * native fun internal_fmt_utf8
+ *
+ *   gas cost: base_cost + unit_cost * bytes_to_return
+ *
+ **************************************************************************************************/
+#[derive(Debug, Clone)]
+pub struct FmtUtf8GasParameters {
+    pub base: InternalGas,
+    pub per_byte: InternalGasPerByte,
+}
+
+fn native_fmt_utf8(
+    gas_params: &FmtUtf8GasParameters,
+    _context: &mut NativeContext,
+    _ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    debug_assert!(args.len() == 1);
+
+    // pop value
+    let ref_to_val = pop_arg!(args, Reference);
+    let val = ref_to_val.read_ref()?;
+    let v_str = &val.to_string()[..];
```

Comment:
The suggestion from @sblackshear was to use existing debug_print functionality for Move. Rust to_string might not be the right thing here.

Independent of that lets first agree on the issue whether we actually want this.

Reply:
- +1 to attempting to reuse the `debug_print` logic
- One way we could proceed is to start by defining this function, but gating it with `#[test]`. I suspect that including this as test-only functionality is fairly non-controversial, but opening it up as a production function will require a lot more scrutiny. I'd be very curious to hear @alinush's opinion on this

Reply:
Hey Sam, when I did the [debug print PR](https://github.com/move-language/move/pull/630), I wrote it so that one could also call it with `canonicalize = true` in a non-testing native function (e.g., a native for canonically-serializing any Move value for signing).

I am no expert on the VM internals though, so I am merely hoping it would be safe to do this.

For example, I hope there are no safety issues around my PR having exposed a `pub fn as_move_value(&self, layout: &MoveTypeLayout) -> MoveValue` function in `Value`.

I also noticed I made a `ValueImpl` function public too, which perhaps was unnecesary. See [here](https://github.com/move-language/move/pull/630/files#diff-36a1671e9470ce7df72a77413b182e251dbe7df3ccaf7a46b5016e045a8cdac9R3535)

Reply:
>Hey Sam, when I did the debug print PR, I wrote it so that one could also call it with canonicalize = true in a non-testing native function (e.g., a native for canonically-serializing any Move value for signing).

Got it, that is helpful.

> For example, I hope there are no safety issues around my PR having exposed a pub fn as_move_value(&self, layout: &MoveTypeLayout) -> MoveValue function in Value.

None at all!